asllib/Parser.mly

```diff
@@ -450,11 +454,13 @@ let local_decl_keyword :=
   | VAR       ; { LDK_Var       }
   *)
 
-let storage_keyword ==
+let storage_keyword :=
```

Comment:
I mirrored `local_decl_keyword` which does the same thing.  I got the comment from it as well.

I've updated the comment to clarify which productions were conflicting for both this case and the E_Record cases.
I haven't renamed this production yet. I'm not actually sure what a good name for it would be and whatever we change it to might want to be mirrored in `local_decl_keyword`. We could add `non_var_..` as a prefix? That feels a bit noisy to me.

Reply:
I've appended `_non_var` to the end of them and updated their corresponding macro renderings in the tex as well as their `build_<name>` functions for ast construction for consistency.

I haven't renamed the macro rules themselves and I haven't touched any adjacent `ASTRule` titles (e.g. `ASTRule.SotrageKeyword`). Should I update those as well or is updating the production name sufficient?